### PR TITLE
feat: Helm chart contribution for argocd-notifications

### PR DESCRIPTION
**Pull request by andyfeller**
The following chart is a contribution from work done at https://github.com/Bandwidth/ to facilitate installing https://github.com/argoproj-labs/argocd-notifications via Helm chart based on https://github.com/argoproj-labs/argocd-notifications/blob/master/manifests/install.yaml

Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by andyfeller** [comment]
@alexmt : I didn't know what the intentions are for https://github.com/argoproj-labs/argocd-notifications/ with regards to Helm charts so feel free to let me know how this contribution needs to be refactored.

**Comment by andyfeller** [comment]
@alexec : I'm not sure who else is appropriate for a reviewer if not you or @alexmt but would appreciate guidance on how helm support for argoproj and argolab repositories is meant to be handled.

**Comment by alexec** [comment]
@andyfeller are you willing to be the maintainer for this chart? If so you should add yourself to the `CODEOWNERS` file. The core team don't maintain these charts - so you might wish to find another maintainer to help you?

File: charts/argocd-notifications/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {{- include "argocd-notifications.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "argocd-notifications.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "argocd-notifications.selectorLabels" . | nindent 8 }}
+    spec:
```
Comment (alexmt) [inline charts/argocd-notifications/templates/deployment.yaml:15]: Please add Recreate strategy to make sure two controllers are not sending notifications during upgrade:

```yaml
  strategy:
    type: Recreate
```

**Comment by andyfeller** [comment]
> @andyfeller are you willing to be the maintainer for this chart? If so you should add yourself to the `CODEOWNERS` file. The core team don't maintain these charts - so you might wish to find another maintainer to help you?

Will be happy to give @alexmt a hand wrt to this

**Comment by andyfeller** [comment]
@alexmt : after this is successfully reviewed and accepted, then I can follow up to open up a PR to include Helm documentation to the argocd-notification project itself

File: CODEOWNERS [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 /charts/argo-events @jbehling
 
-# Argo Workflows
-/charts/argo @benjaminws @stefansedich @paguos
-
 # Argo Rollouts
 /charts/argo-rollouts @cabrinha
+
+# Argo CD Notifications
+/charts/argocd-notifications @alexmt @andyfeller
```
Comment (johnkost) [inline CODEOWNERS:16]: Should this list match the list defined in `Chart.yaml`?
https://github.com/argoproj/argo-helm/pull/365/files#diff-c10017cc8b44885eb18634170c26c1aeR14

**Pull request merged**